### ISSUE TITLE
Automated way to expose configuration schema

### ISSUE THREAD

**Issue by yurishkuro** [open-telemetry/opentelemetry-collector#1995]
## Important (read before submitting)

**Is your feature request related to a problem? Please describe.**

Unless I am missing something, there is no authoritative way to generate documentation for the schema of configuration file. There is some manually maintained doc pages with examples, but the authoritative source is the source code of the config structs (e.g. [pprof extension](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/pprofextension/config.go)). This is pretty difficult for end-users.

For Jaeger binaries built on top of OTEL collector, we could like to do better than that. For example, the current Jaeger configuration can be inspected from the `help` sub-commands and/or auto-generated docs like https://www.jaegertracing.io/docs/1.20/cli/.

**Describe the solution you'd like**

A way to auto-generate readable documentation for configuration options, which means:
  * parseable authoritative source of the schema with inline documentation, such as `.proto` definition or even Go structs
  * a utility that can convert that source of schema to another format (preferable something structured like JSON or YAML), so that it can be included in the HTML docs on a website.

**Comment by jrcamp** [comment]
Definitely. We do this in the SignalFx agent by using the comment associated with the config property along with Go tags for things like default and validation. See https://github.com/signalfx/signalfx-agent/blob/master/pkg/monitors/ntp/ntp.go#L30 for example. I think a similar approach would work here.

**Comment by yurishkuro** [comment]
@jrcamp +1, struct tags can be a viable way to provide some metadata, although including long description into the tags is pretty inconvenient, those are better as field comments. Did you implement some utility that can extract the schema from structs? To read both tags and comments the tool would be similar to godoc, working off the AST.

**Comment by bogdandrutu** [comment]
@jrcamp can we start configuring this for one of the component, and see how will look like?

**Comment by jrcamp** [comment]
I've filed a separate issue #2015 for config validation. Could have some impact on this but don't think it blocks anything, just something to be aware of.
